### PR TITLE
bpo-46039: Split yield from in two

### PR DESCRIPTION
**Pull request by markshannon**
Splits `YIELD_FROM` into `SEND` and `JUMP_ABSOLUTE`. This is part of our push to make all operations explicit in the bytecode for instrumentation and optimization.

This is a partial implementation of [bpo-46039](https://bugs.python.org/issue46039).
It does not do the threeway split described in issue, nor use zero-cost exception handling to replace the special case case in gen.throw().

https://bugs.python.org/issue46039

### CODE REVIEW

File: Objects/frameobject.c [hunk 1]
```diff
@@ -246,7 +246,13 @@ mark_stacks(PyCodeObject *code_obj, int len)
                     next_stack = pop_value(pop_value(pop_value(next_stack)));
                     stacks[i+1] = next_stack;
                     break;
-
+                case SEND:
+                    j = get_arg(code, i) + i + 1;
+                    assert(j < len);
+                    assert(stacks[j] == UNINITIALIZED || stacks[j] == pop_value(next_stack));
```
Comment (lpereira) [inline Objects/frameobject.c:255]: Is this correct?  Assertions will be removed in release builds and `pop_value()` has side-effects.

File: Lib/importlib/_bootstrap_external.py [hunk 2]
```diff
@@ -380,7 +381,7 @@ def _write_atomic(path, data, mode=0o666):
 # Whenever MAGIC_NUMBER is changed, the ranges in the magic_values array
 # in PC/launcher.c must also be updated.
 
-MAGIC_NUMBER = (3465).to_bytes(2, 'little') + b'\r\n'
+MAGIC_NUMBER = (3467).to_bytes(2, 'little') + b'\r\n'
```
Comment (iritkatriel) [inline Lib/importlib/_bootstrap_external.py]:
```suggestion
MAGIC_NUMBER = (3466).to_bytes(2, 'little') + b'\r\n'
```

File: Python/compile.c [hunk 3]
```diff
@@ -1804,6 +1806,24 @@ compiler_call_exit_with_nones(struct compiler *c) {
     return 1;
 }
 
+static int
+compiler_add_yield_from(struct compiler *c)
+{
+    basicblock *start, *jump, *exit;
```
Comment (lpereira) [inline Python/compile.c:1827]: Would be nice if `exit` had another name, as this shadows the `exit()` function

File: Objects/frameobject.c (hunk 1 above)
Comment (markshannon) [reply Objects/frameobject.c:255]: No side effects on `int`s.

Comment (lpereira) [reply Objects/frameobject.c:255]: (Disregard the comment. I should have looked at what `pop_value()` does.)

Comment (arhadthedev) [reply Objects/frameobject.c:255]: Well, `pop_value` name implies that it modifies a stack (I suspect that's what @lpereira means as a side effect). In reality though it's rather `get_stack_top`:

https://github.com/python/cpython/blob/d4363d214097b3fca8b7910c2e0e91c8b0873fb2/Objects/frameobject.c#L167-L171

https://github.com/python/cpython/blob/cb8f491f46e262549f6c447b31625cab7c20a60a/Include/pyport.h#L296-L301

**Pull request merged**